core/dbt/task/run.py

```diff
@@ -284,18 +374,95 @@ def execute(self, model, manifest):
             )
 
         hook_ctx = self.adapter.pre_model_hook(context_config)
+        batch_results = None
         try:
-            result = MacroGenerator(
-                materialization_macro, context, stack=context["context_macro_stack"]
-            )()
+            if (
+                os.environ.get("DBT_EXPERIMENTAL_MICROBATCH")
+                and model.config.materialized == "incremental"
+                and model.config.incremental_strategy == "microbatch"
+            ):
+                batch_results = self._execute_microbatch_materialization(
+                    model, manifest, context, materialization_macro
+                )
+            else:
+                result = MacroGenerator(
+                    materialization_macro, context, stack=context["context_macro_stack"]
+                )()
+                for relation in self._materialization_relations(result, model):
+                    self.adapter.cache_added(relation.incorporate(dbt_created=True))
```

Comment:
I know we talked about this work, and I absolutely understand why we're doing it this way. Something feels smelly about it though, and I can't exactly put my finger on it. My best guess is that we're doing a conditional, exiting the conditional, and then basically re-entering the conditional on lines 396-399. For instance line 399 should _never_ be hit if we enter line 384. However because of the split conditionals, this isn't immediately apparent. I wonder if we should be calling into two separate private functions just before the `try` on line 378, and only one or the other function would ever be called depending on if we're doing microbatch stuff or not.